pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -68,6 +71,7 @@
  */
 public class ProxyConnection extends PulsarHandler implements FutureListener<Void> {
     // ConnectionPool is used by the proxy to issue lookup requests
+    private static volatile Timer shareTimer = null;
```

Comment:
can we add this variable as regular field inside ProxyService ?
this way:
- you do not need to use a static variable
- you have a clean lifecycle for the sharedTime (you will shutdown it when you shutdown the ProxyService)